onnxruntime/python/ort_trainer.py

```diff
@@ -10,7 +10,7 @@
 from distutils.version import LooseVersion
 
 class IODescription():
-    def __init__(self, name, shape, dtype, num_classes=None):
+    def __init__(self, name, shape, dtype=None, num_classes=None):
```

Comment:
so with this PR, user code will not have to specify data types. Is there other benefit? Maybe to update onnxruntime_test_ort_trainer to reflect these benefit.

Reply:
Yes tests should be updated to reflect the changes. The issue is that current test relies on dtype and vocab size to generate sample data.